benchmarks/ccl_test_power_MG.c

```diff
@@ -0,0 +1,218 @@
+		data->n_s,-1,-1,-1,data->mu_0[i_model], data->sigma_0[i_model],-1,NULL,NULL, &status);
+
+  ccl_cosmology * cosmo= ccl_cosmology_create(params, config_linear);
+  ASSERT_NOT_NULL(cosmo);
+  
+  sprintf(fname,"./tests/benchmark/model%d_pk_MG.dat",i_model);
+   f=fopen(fname,"r");
+  if(f==NULL) {
+    fprintf(stderr,"Error opening file %s\n",fname);
+    exit(1);
+  }
+  nk=linecount(f)-1; rewind(f);
+  
+  double k=0.,pk_bench=0.,pk_ccl,err, k_h, pk_h;
+  double z=0.; //Other redshift checks are possible 
+  int stat=0;
+
+	for(i=0;i<nk;i++) {      
+    stat=fscanf(f,"%le %le\n",&k_h, &pk_h);
+    if(stat!=2) {
+      fprintf(stderr,"Error reading file %s, line %d\n",fname,i);
+      exit(1);
+    }
+     
+    // This is required if benchmark is in little h units. (Mpc/h etc)
+    k=k_h*data->h;
+    pk_bench=pk_h/pow(data->h,3);
+    
+    pk_ccl=ccl_linear_matter_power(cosmo,k,1./(1+z),&status);
+    
+    err=fabs(pk_ccl/pk_bench-1);
+    ASSERT_DBL_NEAR_TOL(err,0.,POWER_MG_TOL);
+    }
+    
+  fclose(f);
+
+  ccl_cosmology_free(cosmo);
+
+}
+
+static void check_transfer_error(ccl_configuration config, struct power_MG_data * data)
+{
+  int status=0;
+  
+  ccl_parameters params;
+  
+  // Initialize ccl_cosmology struct
+  params = ccl_parameters_create(data->Omega_c, data->Omega_b, data->Omega_k,
+		data->Neff, &(data->mnuval), data-> mnu_type, 
+		data->w0, data->wa,  data->h, data->A_s, 
+		data->n_s,-1,-1,-1,data->mu_0[0], data->sigma_0[0],-1,NULL,NULL, &status);
+
+  ccl_cosmology * cosmo= ccl_cosmology_create(params, config);
+  ASSERT_NOT_NULL(cosmo);
+  
+  // Call P(k) with unacceptable transfer function methods, check we get expected error.
+  ccl_cosmology_compute_power(cosmo, &status);
+  ASSERT_STR(cosmo->status_message, "ccl_power.c: ccl_cosmology_compute_power(): The power spectrum in the mu / Sigma modified gravity parameterisation is only implemented with the ccl_boltzmann_class power spectrum method.\n");
+  
+  ccl_cosmology_free(cosmo);
+}
+
+static void check_nonlin_error(struct power_MG_data * data)
+{
+  int status=0;
+  
+  ccl_configuration config = default_config;
+  
+  ccl_parameters params;
+  
+  // Initialize ccl_cosmology struct
+  params = ccl_parameters_create(data->Omega_c, data->Omega_b, data->Omega_k,
+		data->Neff, &(data->mnuval), data-> mnu_type, 
+		data->w0, data->wa,  data->h, data->A_s, 
+		data->n_s,-1,-1,-1,data->mu_0[0], data->sigma_0[0],-1,NULL,NULL, &status);
+
+  ccl_cosmology * cosmo= ccl_cosmology_create(params, config);
+  ASSERT_NOT_NULL(cosmo);
+  
+  // Call P(k) trying to get the nonlinear power spectrum, check we get expected error.
+  double k = 0.1;
+  double a = 1.;
+  double pk = ccl_nonlin_matter_power(cosmo, k, a, &status);
+  ASSERT_STR(cosmo->status_message, "ccl_power.c: ccl_nonlin_matter_power(): Nonlinear behaviour for the mu / Sigma parameterization of modified gravity is not implemented. \n");
+  
+  ccl_cosmology_free(cosmo);
+}
+
+/*
```

Comment:
Good question - the last four tests are commented out because of the ongoing issues with understanding how well we expect our code to match the benchmark produced by the UT group at larger k. These I expect we will re-include before we merge with appropriate tolerance for discrepancy at appropriate k (the UT group is actively working on figuring this out at I understand). The first four tests, however, were just commented out in error; I included them again.